tests/arrow_record_batch_gen/mod.rs

```diff
@@ -277,30 +276,63 @@ pub(crate) fn get_arrow_date_record_batch() -> (RecordBatch, SchemaRef) {
 
 // struct
 pub(crate) fn get_arrow_struct_record_batch() -> (RecordBatch, SchemaRef) {
-    let boolean = Arc::new(BooleanArray::from(vec![false, false, true, true]));
-    let int = Arc::new(Int32Array::from(vec![42, 28, 19, 31]));
-
-    let struct_array = StructArray::from(vec![
-        (
-            Arc::new(Field::new("b", DataType::Boolean, false)),
-            boolean.clone() as ArrayRef,
-        ),
-        (
-            Arc::new(Field::new("c", DataType::Int32, false)),
-            int.clone() as ArrayRef,
-        ),
-    ]);
-
     let schema = Arc::new(Schema::new(vec![Field::new(
         "struct",
-        DataType::Struct(Fields::from(vec![
-            Field::new("b", DataType::Boolean, false),
-            Field::new("c", DataType::Int32, false),
-        ])),
-        false,
+        DataType::Struct(
+            vec![
+                Field::new("b", DataType::Boolean, true),
+                Field::new("c", DataType::Int32, true),
+            ]
+            .into(),
+        ),
+        true,
     )]));
 
-    let record_batch = RecordBatch::try_new(Arc::clone(&schema), vec![Arc::new(struct_array)])
+    let mut struct_builder = StructBuilder::new(
+        vec![
+            Field::new("b", DataType::Boolean, true),
+            Field::new("c", DataType::Int32, true),
+        ],
+        vec![
+            Box::new(BooleanBuilder::new()),
+            Box::new(Int32Builder::new()),
+        ],
+    );
+
+    struct_builder
+        .field_builder::<BooleanBuilder>(0)
+        .expect("should return field builder")
+        .append_value(false);
+    struct_builder
+        .field_builder::<Int32Builder>(1)
+        .expect("should return field builder")
+        .append_value(30);
+    struct_builder.append(true);
+
+    // NULL struct item is temporary disabled as not properly supported by duckdb and postgres
```

Comment:
Updated get_arrow_struct_record_batch in order to be able to include test for None struct value. Tested for sqlite (works) but temporary disabled as duckdb and postgres don't handle it correctly. To be investigated.